tests/test_coverage.py

```diff
@@ -53,3 +53,30 @@ def test_coverage_symmetric_upper(request):
     assert (tot_cov == np.array([3, 1, 2])).all()
     assert clr.info["cis"] == 1
     assert clr.info["sum"] == 3
+    
+def test_coverage_weight(request):
+    # perform test:
+    clr = cooler.Cooler(op.join(request.fspath.dirname, "data/CN.mm9.1000kb.cool"))
+    cis_cov_weight, tot_cov_weight = cooltools.api.coverage.coverage(
+        clr, ignore_diags=2, chunksize=int(1e7), clr_weight_name="weight"
+    )
+    
+    # Test that mean total balanced coverage is 1.0
+    assert np.nanmean(tot_cov_weight) == 1.0
+    
+    # Generate test matrix with weights
+    bins = pd.DataFrame(
+        [["chr1", 0, 1, 0.5], ["chr1", 1, 2, 1], ["chrX", 1, 2, 0.2]],
+        columns=["chrom", "start", "end", "weight"],
+    )
+
+    pixels = pd.DataFrame(
+        [[0, 1, 1], [0, 2, 2]], columns=["bin1_id", "bin2_id", "count"]
+    )
+
+    clr_file = op.join(request.fspath.dirname, "data/test_coverage.cool")
+    cooler.create_cooler(clr_file, bins, pixels)
+    clr = cooler.Cooler(clr_file)
+    cis_cov_weight, tot_cov_weight = cooltools.coverage(clr, ignore_diags=0, store=True, clr_weight_name="weight")
+    assert (cis_cov_weight == np.array([0.5, 0.5, 0])).all()
```

Comment:
also for tests - should we add a bin or 2 with `NaN` in weights - just to make sure that bias masking mechanism is working ok  - in case we ever break it ...

Reply:
YES to adding bin with Nans
it would also be good to add test that covers the usage of the prefixes & storing

Reply:
The storing to cooler part, I believe is tested - in the test function above this one

Reply:
Yes good idea with allclose and nans. Implemented this, let me know!